src/json.c

```diff
@@ -101,17 +62,98 @@ int FieldSpec_CheckJsonType(FieldType fieldType, JSONType type) {
     break;
   // Boolean values can be represented only as TAG 
   case JSONType_Bool:    
-  // An array can be represented only as TAG 
+  // An array can be represented only as TAG
+  case JSONType_Array:
     if (fieldType == INDEXFLD_T_TAG) {
       rv = REDISMODULE_OK;
     }
     break;
   // An object or null type are not supported
-  case JSONType_Array: // TODO: array will be supported as TAG
   case JSONType_Object:
   case JSONType_Null:
   case JSONType__EOF:
     break;
   }
   return rv;
-}
\ No newline at end of file
+}
+
+int JSON_GetRedisModuleString(RedisModuleCtx *ctx, JSONResultsIterator jsonIter,
+                              FieldType ftype, struct DocumentField *df) {
+  RedisJSON json = japi->next(jsonIter);
+  if (!json) {
+    RS_LOG_ASSERT(0, "shouldn't happen");
+  }
+
+  if (FieldSpec_CheckJsonType(ftype, japi->getType(json))) {
+    return REDISMODULE_ERR;
+  }
+  
+  int boolval;
+  long long intval;
+  const char *str;
+  size_t len;
+  int rv = REDISMODULE_OK;
+
+  switch (japi->getType(json)) {
+    case JSONType_String:
+      japi->getString(json, &str, &df->strlen);
```

Comment:
Where do you verify that you do not try to index a string as a number? our pre-processor will translate it no?

Reply:
The switch is on `JSONType` returned by `japi->getType()`.
At `FieldSpec_CheckJsonType` the type is verified against the field.